components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/GatewayArtifactsMgtDAO.java

```diff
@@ -0,0 +1,219 @@
+                    String gatewayRuntimeArtifacts = IOUtils.toString(inputStream,
+                            APIConstants.DigestAuthConstants.CHARSET);
+                    gatewayRuntimeArtifactsArray.add(gatewayRuntimeArtifacts);
+                } catch (IOException  e) {
+                    handleException("Error in generating gatewayRuntimeArtifacts ", e);
+                }
+            }
+        } catch (SQLException e) {
+            handleException("Failed to get artifacts " , e);
+        }
+        return gatewayRuntimeArtifactsArray;
+    }
+
+    /**
+     * Check whether the API is published in any of the Gateways
+     *
+     * @param APIId - UUID of the API
+     * @throws APIManagementException if an error occurs
+     */
+    public boolean isAPIPublishedInAnyGateway(String APIId) throws APIManagementException {
+
+        int count = 0;
+        try (Connection connection = GatewayArtifactsMgtDBUtil.getArtifactSynchronizerConnection();
+                PreparedStatement statement = connection.prepareStatement(SQLConstants.GET_PUBLISHED_GATEWAYS_FOR_API)) {
+            statement.setString(1, APIId);
+            statement.setString(2, APIConstants.GatewayArtifactSynchronizer.GATEWAY_INSTRUCTION_PUBLISH);
+            ResultSet rs = statement.executeQuery();
+            while (rs.next()) {
+                count = rs.getInt("COUNT");
+            }
+        } catch (SQLException e) {
+            handleException("Failed check whether API is published in any gateway " + APIId, e);
+        }
+        return count != 0;
+    }
+
+    /**
+     * Check whether the API details exists in the db
+     *
+     * @param APIId - UUID of the API
+     * @throws APIManagementException if an error occurs
+     */
+    public boolean isAPIDetailsExists(String APIId) throws APIManagementException {
+
+        try (Connection connection = GatewayArtifactsMgtDBUtil.getArtifactSynchronizerConnection();
+                PreparedStatement statement = connection
+                        .prepareStatement(SQLConstants.GET_GATEWAY_PUBLISHED_API_DETAILS)) {
+            statement.setString(1, APIId);
+            ResultSet rs = statement.executeQuery();
+            return rs.next();
+        } catch (SQLException e) {
+            handleException("Failed to check API details status of API with ID " + APIId, e);
+        }
+        return false;
+    }
+
+    /**
+     * Check whether the API artifact for given label exists in the db
+     *
+     * @param APIId - UUID of the API
+     * @throws APIManagementException if an error occurs
+     */
+    public boolean isAPIArtifactExists(String APIId, String gatewayLabel) throws APIManagementException {
+
+        int count = 0;
+        try (Connection connection = GatewayArtifactsMgtDBUtil.getArtifactSynchronizerConnection();
+                PreparedStatement statement = connection.prepareStatement(SQLConstants.CHECK_ARTIFACT_EXISTS)) {
```

Comment:
make the query Select 1 and check if rs.next

Reply:
https://github.com/wso2/carbon-apimgt/pull/8828/commits/6604c7cf1386950cd25886fffa21addbbc595804